### PR TITLE
Spring clean/a d

### PR DESCRIPTION
**Pull request by tomalrussell**
Rebased after https://github.com/nismod/smif/pull/33 so should be clean.

- remove Controller, rename controller.py to sos_model.py
- move data reading and validation to smif.data_layer
- work on test coverage, including running models with scenario data

### CODE REVIEW

**Comment by tomalrussell** [comment]
@willu47 ready for review

File: smif/data_layer/sector_model_config.py [hunk 1]
```diff
@@ -4,34 +4,55 @@
...
-    model_name : str
-        The name of the model
-    model_path : str
-        The path to the python module file that contains an implementation
-        of SectorModel
-    model_classname : str
-        The name of the class that implements SectorModel
-    model_config_dir : str
-        The root path of model config/data to use
+    initial_config : dict
```
Comment (willu47) [inline smif/data_layer/sector_model_config.py]: Nice docstrings!  The following format makes each of the keys here a mini-heading::
```python
    initial_config: dict
        Sector model details, sufficient to read the full config from a set of
        files. Must contain the following fields:

            "model_name"
                The name of the sector model, for reference within the
                system-of-systems model

            "model_path"
                The path to the python module file that contains an
                implementation of SectorModel

            "model_classname"
                The name of the class that implements SectorModel

            "model_config_dir"
                The root path of model config/data to use, which
                must contain inputs.yaml, outputs.yaml, time_intervals.yaml and
                regions.shp/regions.geojson

            "initial_conditions"
                List of files containing initial conditions

            "interventions"
                List of files containing interventions
```

File: tests/data_layer/test_sector_model_config.py [hunk 2]
```diff
@@ -43,22 +50,22 @@ def test_load_errors(self, setup_project_missing_model_config):
         reader = self._reader(setup_project_missing_model_config)
 
         with raises(FileNotFoundError) as ex:
-            reader._load_inputs()
```
Comment (willu47) [inline tests/data_layer/test_sector_model_config.py]: Brownie points for these tests!

File: tests/data_layer/test_sos_model_config.py [hunk 3]
```diff
@@ -69,10 +82,33 @@ def test_timesteps_alternate_file(self, setup_project_folder,
         actual = reader.timesteps
         assert actual == expected
 
-    def test_timesteps_invalid(self, setup_project_folder,
-                               setup_timesteps_file_invalid):
-
-        reader = SosModelReader(self._get_model_config(setup_project_folder))
+    def test_timesteps_invalid(self, setup_project_folder, setup_timesteps_file_invalid):
+        """Expect an error on trying to load an invalid file
```
Comment (willu47) [inline tests/data_layer/test_sos_model_config.py]: These comments are useful

File: smif/sos_model.py [hunk 4]
```diff
@@ -4,56 +4,18 @@
...
-        Parameters
-        ----------
-        model_name: str
-            The name of the sector model to run
-        """
-        self._model.run_sector_model(model_name)
-
-
 class SosModel(object):
     """Consists of the collection of timesteps and sector models
```
Comment (willu47) [inline smif/sos_model.py]: Need to update the refererences to `Controller` in the docstrings to the SosModel class

File: smif/data_layer/sector_model_config.py (hunk 1 above)
Comment (tomalrussell) [inline smif/data_layer/sector_model_config.py]: Great - ae972ef should do the trick.

File: smif/sos_model.py (hunk 4 above)
Comment (tomalrussell) [inline smif/sos_model.py]: Updated with 	c162b08

**Pull request merged**